extensions/dicom-microscopy/src/DicomMicroscopyViewport.tsx

```diff
@@ -1,318 +1,217 @@
 import getDicomWebClient from './utils/dicomWebClient';
 import dcmjs from 'dcmjs';
-import MicroscopyService from './services/MicroscopyService';
-import { CustomizationService } from '@ohif/core';
-
-class DicomMicroscopyViewport extends Component {
-  state = {
-    error: null as any,
-    isLoaded: false,
-  };
-
-  microscopyService: MicroscopyService;
-  customizationService: CustomizationService;
-  viewer: any = null; // dicom-microscopy-viewer instance
-  managedViewer: any = null; // managed wrapper of microscopy-dicom extension
-
-  container = React.createRef();
-  overlayElement = React.createRef();
-
-  constructor(props: any) {
-    super(props);
-
-    const { microscopyService, customizationService } = this.props.servicesManager.services;
-    this.microscopyService = microscopyService;
-    this.customizationService = customizationService;
-  }
-
-  static propTypes = {
-    viewportData: PropTypes.object,
-    activeViewportId: PropTypes.string,
-    setViewportActive: PropTypes.func,
-
-    // props from OHIF Viewport Grid
-    displaySets: PropTypes.array,
-    viewportId: PropTypes.string,
-    viewportLabel: PropTypes.string,
-    dataSource: PropTypes.object,
-    viewportOptions: PropTypes.object,
-    displaySetOptions: PropTypes.array,
-
-    // other props from wrapping component
-    servicesManager: PropTypes.object,
-    extensionManager: PropTypes.object,
-    commandsManager: PropTypes.object,
-    resizeRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({ current: PropTypes.any })]),
-  };
-
-  /**
-   * Get the nearest ROI from the mouse click point
-   *
-   * @param event
-   * @param autoselect
-   * @returns
-   */
-  getNearbyROI(event: Event, autoselect = true) {
-    const symbols = Object.getOwnPropertySymbols(this.viewer);
-    const _drawingSource = symbols.find(p => p.description === 'drawingSource');
-    const _pyramid = symbols.find(p => p.description === 'pyramid');
-    const _map = symbols.find(p => p.description === 'map');
-    const _affine = symbols.find(p => p.description === 'affine');
-
-    const feature = this.viewer[_drawingSource].getClosestFeatureToCoordinate(
-      this.viewer[_map].getEventCoordinate(event)
-    );
-
-    if (!feature) {
-      return null;
-    }
-
-    const roiAnnotation = this.viewer._getROIFromFeature(
-      feature,
-      this.viewer[_pyramid].metadata,
-      this.viewer[_affine]
-    );
-    if (roiAnnotation && autoselect) {
-      this.microscopyService.selectAnnotation(roiAnnotation);
-    }
-    return roiAnnotation;
-  }
+import { useAppConfig } from '@state';
+
+function DicomMicroscopyViewport({
+  activeViewportId,
+  setViewportActive,
+  displaySets,
+  viewportId,
+  dataSource,
+  servicesManager,
+  extensionManager,
+  resizeRef,
+}: {
+  activeViewportId: string;
+  setViewportActive: Function;
+  displaySets: any[];
+  viewportId: string;
+  dataSource: any;
+  servicesManager: any;
```

Comment:
Since you are changing things, please REMOVE the services and extension manager from the API and useSystem instead 
That is,
   const { servicesManager, extensionManager } = useSystem()

Reply:
You will also need to remove the custom version in the getPanelModule, so that it just directly returns DicomMicroscopyViewport.